keel-lemur/src/main/kotlin/com/netflix/spinnaker/keel/lemur/LemurCertificateResolver.kt

```diff
@@ -0,0 +1,49 @@
+package com.netflix.spinnaker.keel.lemur
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.ApplicationLoadBalancerSpec
+import com.netflix.spinnaker.keel.api.ec2.EC2_APPLICATION_LOAD_BALANCER_V1_2
+import com.netflix.spinnaker.keel.api.plugins.Resolver
+import com.netflix.spinnaker.kork.exceptions.ConstraintViolationException
+import kotlinx.coroutines.runBlocking
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnBean(LemurService::class)
+class LemurCertificateResolver(
+  private val lemurService: LemurService
+) : Resolver<ApplicationLoadBalancerSpec> {
+  override val supportedKind = EC2_APPLICATION_LOAD_BALANCER_V1_2
+
+  override fun invoke(resource: Resource<ApplicationLoadBalancerSpec>): Resource<ApplicationLoadBalancerSpec> =
+    resource.copy(
+      spec = resource.spec.copy(
+        listeners = resource.spec.listeners.mapTo(mutableSetOf()) { listener ->
+          listener.certificate?.let { name ->
+            listener.copy(
+              certificate = runBlocking { findCurrentCertificate(name) }
+            )
+          } ?: listener
+        }
+      )
+    )
+
+  private suspend fun findCurrentCertificate(name: String): String {
+    val certificate = lemurService.certificateByName(name).items.first()
+    return if (certificate.active) {
+      certificate.name
+    } else {
+      val replacement = certificate.replacedBy.firstOrNull { it.active }
+      replacement?.name
+        ?: if (certificate.replacedBy.isNotEmpty()) {
+          findCurrentCertificate(certificate.replacedBy.first().name)
+        } else {
+          throw CertificateExpired(certificate)
+        }
+    }
+  }
```

Comment:
Find the cert by name, and return it if it's active, otherwise return the replacement if _that_ is active. If necessary recurse to look up further replacements.

Reply:
@luispollo I revised the certificate lookup so that it throws a proper exception, and if necessary recurses to find the current certificate